app/controllers/products_controller.rb

```diff
@@ -2,4 +2,33 @@ class ProductsController < ApplicationController
   def index
     @products = Product.all
   end
+
+  before_action :authenticate_admin!, only: [:edit]
```

Comment:
put this on line two and write it like this instead:

`before_action :authenticate_admin!, only: [:update, :destroy, :edit]`

Reply:
Changes made accordingly.